### DISCUSSION TITLE
Implementation of modal items

### DISCUSSION THREAD

**Discussion by ascpial** [nextcord/nextcord#443] [General]
DISCLAIMER: this is my first contribution and i'm not very confident with the library yet. Also, I am not very good in English

I'm trying to implement the modal feature in nextcord.
I implemented textinputs in the same way as buttons and selects, but now I'm wondering the best way to implemente modals in the library.
I was thinking that I can had modal like a View, but I don't know if I need to subclass or rewrite a Modal class from scratch.
I can also add a feature that allow me to send a view as an interaction response (e.g. `interaction.send_modal(<some subclass of nextcord.View>)`)

**Comment by DenverCoder1** [comment]
As it is a type of interaction response, I believe it should be under `InteractionResponse.send_modal()` and due to the similarity between views and modals, I agree that Modals could be a subclass of View.
